floats/floats.go

```diff
@@ -923,3 +923,22 @@ func Within(s []float64, v float64) int {
 	}
 	return -1
 }
+
+// NeumaierSum returns the sum of the elements of the slice, calculated in a more
+// accurate but also more expensive way than Sum. It uses an improved version of
+// Kahan's compensated summation algorithm proposed by Neumaier: see
+// https://en.wikipedia.org/wiki/Kahan_summation_algorithm for details.
+func NeumaierSum(s []float64) float64 {
```

Comment:
I'd suggest considering calling this `SumCompensated` so that we are not constrained by the algorithm if we decide at a later stage to change it. This would the become
```
// SumCompensated returns the sum of the elements of the slice calculated with greater
// accuracy than Sum at the expense of additional computation.
func SumCompensated(s []float64) float64 {
	// SumCompensated uses an improved version of Kahan's compensated
	// summation algorithm proposed by Neumaier.
	// See https://en.wikipedia.org/wiki/Kahan_summation_algorithm for details.
```

Reply:
I'd prefer this wording and arrangement please.

Reply:
Sorry, missed that. Will do.